editor/src/core/workers/utils.ts

```diff
@@ -1,10 +1,43 @@
-import { BASE_URL, URL_HASH } from '../../common/env-vars'
-import { setBranchNameFromURL } from '../../utils/branches'
-
-export function workerForFile(fileName: string): Worker {
-  let workerURL = new URL(`${BASE_URL}${fileName}`)
-  let workerQueryParams = workerURL.searchParams
-  workerQueryParams.set('hash', URL_HASH)
-  setBranchNameFromURL(workerQueryParams)
-  return new Worker(workerURL.href)
+import { BASE_URL } from '../../common/env-vars'
+
+/**
+ * Webpack 5 only supports a single public path for assets and feeding the Worker url via `import.meta.url`
+ * This causes problems when using a CDN, because of same origin restrictions, you are not allowed to load workers from cdns.
+ *
+ * The solution is to use the webpack-supported global __webpack_public_path__ variable, which updates the value returned by import.meta.url
+ *
+ * in all of these cases, oldPublicPath is the asset url (the CDN url in deployed editors). we temporarily change it to BASE_URL (which is the url the editor is loaded from)
+ * and then after loading the worker, restore it to the CDN url.
+ */
+
+export function createTsWorker(): Worker {
+  const oldPublicPath = __webpack_public_path__
+  __webpack_public_path__ = BASE_URL
```

Comment:
Did you try using the [`publicPath`](https://webpack.js.org/loaders/worker-loader/#publicpath) config option?

Reply:
that is set to our CDN

Reply:
You can provide a separate value for the worker loader via the rules (that link shows a couple of examples)

Reply:
the worker loader is for webpack 4, it has been superseded by the `new Worker(...` syntax in webpack 5, which has no configuration options

Reply:
:facepalm:

Reply:
yep.